rfcs/20230622-quantized-reduction.md

```diff
@@ -0,0 +1,522 @@
+  at `dimensions`.
+* `input_slices_converted = map(input_conversion, input_slices...)`.
+* `reduce(input_slices_converted) = exec(schedule)` for some binary tree
+  `schedule` where:
+  * `exec(node) = body(exec(node.left), exec(node.right))`.
+  * `exec(leaf) = leaf.value`.
+* `schedule` is an implementation-defined full binary tree whose in-order
+  traversal consists of:
+  * `input_slices_converted...[index]` values, for all `index` in
+    `index_space(input_slices_converted)` in the ascending lexicographic order
+    of `index`.
+  * Interspersed with an implementation-defined amount of `init_values`
+    at implementation-defined positions.
+
+#### Inputs
+
+| Label | Name                | Type                                         | Constraints |
+|-------|---------------------|----------------------------------------------|-------------|
+| (I?)  | `inputs`            | variadic number of tensors                   |             |
+| (I?)  | `init_values`       | variadic number of 0-dimensional tensors     |             |
+| (I?)  | `dimensions`        | 1-dimensional tensor constant of type `si64` |             |
+| (I?)  | `input_conversion`  | function                                     |             |
+| (I?)  | `body`              | function                                     |             |
+| (I?)  | `output_conversion` | function                                     |             |
+
+#### Outputs
+
+| Name      | Type                       | Constraints |
+|-----------|----------------------------|-------------|
+| `results` | variadic number of tensors |             |
+
+#### Constraints
+
+* (C?) `same(shape(inputs...))`.
+* (C?) `element_type(inputs...) = element_type(init_values...)`.
+* (C?) `baseline_element_type(inputs...) = baseline_element_type(results...)`.
+* (C?) `0 < size(inputs) = size(init_values) = size(results) = N`.
+* (C?) `0 <= dimensions < rank(inputs[0])`.
+* (C?) `is_unique(dimensions)`.
+* (C?) `input_conversion` has type `tensor<E0>, ..., tensor<EN-1> ->
+       (tensor<E'0>, ..., tensor<E'N-1>)` where `Ei = element_type(inputs[i])`.
+* (C?) `body` has type `tensor<E0>, ..., tensor<EN-1>, tensor<E0>, ...,`
+       `tensor<EN-1>) -> (tensor<E0>, ..., tensor<EN-1>)` where
+       `Ei = element_type(output_types(input_conversion)[i])`.
+* (C?) `output_conversion` has type `tensor<E0>, ..., tensor<EN-1> ->
+       (tensor<E'0>, ..., tensor<E'N-1>)` where
+       `E'i = element_type(results[i])`.
+* (C?) `element_type(output_types(input_conversion)...) =
+       element_type(input_types(output_conversion)...)`.
+* (C?) `shape(results...) = shape(inputs...)` except that the dimension
+  sizes of `inputs...` corresponding to `dimensions` are not included.
+```
+
+The above specification of `reduce` op can be used to define the specification
+of other ops as shown below. For brevity, we are only presenting the relevant
+portions of the spec which needs modification.
+
+### reduce_window
+
+```python
```

Comment:
Why are you wrapping the following sections with python code block?

Reply:
Initially I thought about isolating the proposed spec of the ops in a code block for better visibility. Now it seems unnecessary and hence removed.